src/Interpreters/DDLTask.cpp

```diff
@@ -215,20 +215,47 @@ ContextMutablePtr DDLTaskBase::makeQueryContext(ContextPtr from_context, const Z
 }
 
 
-bool DDLTask::findCurrentHostID(ContextPtr global_context, Poco::Logger * log, const ZooKeeperPtr & zookeeper)
+bool DDLTask::findCurrentHostID(ContextPtr global_context, Poco::Logger * log, const ZooKeeperPtr & zookeeper, const std::optional<std::string> & config_host_name)
 {
     bool host_in_hostlist = false;
     std::exception_ptr first_exception = nullptr;
 
+    const auto maybe_secure_port = global_context->getTCPPortSecure();
+    const auto port = global_context->getTCPPort();
+
+    if (config_host_name)
+    {
+        bool is_local_port = (maybe_secure_port && HostID(*config_host_name, *maybe_secure_port).isLocalAddress(*maybe_secure_port)) ||
+                             HostID(*config_host_name, port).isLocalAddress(port);
```

Comment:
But it still relies on DNS

```
bool HostID::isLocalAddress(UInt16 clickhouse_port) const
{
    try
    {
        return DB::isLocalAddress(DNSResolver::instance().resolveAddress(host_name, port), clickhouse_port);
```

And although it addresses @tavplubix comment here - https://github.com/ClickHouse/ClickHouse/pull/57603/files#r1426976773 the purpose of this patch is a bit unclear

But why I came here is that I really need behaviour without this extra check, since I indeed want to allow executing queries on the node that does not know anything about all their IP addresses and the one that is used for communication is not added to any interface on the machine, so this check fails

The original idea was using UUIDs, have you considered this? Or maybe you will not be against remove this check?